copy/copy.go

```diff
@@ -814,7 +823,25 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		// If this fails while writing data, it will do pipeWriter.CloseWithError(); if it fails otherwise,
 		// e.g. because we have exited and due to pipeReader.Close() above further writing to the pipe has failed,
 		// we don’t care.
-		go compressGoroutine(pipeWriter, destStream) // Closes pipeWriter
+		go compressGoroutine(pipeWriter, destStream, desiredCompressionFormat, c.compressionLevel) // Closes pipeWriter
```

Comment:
(Does the compression level really need to be modifiable in the first place? Most implementations ignore the value, that’s going to be annoying to document and deal with.)

Reply:
I've seen very significant differences when using higher values with zstd (i.e. ~60M vs ~90M), but higher compression levels are quite CPU intensive.

Also the compression buffer size has a noticeable impact on the compression